app/forms/fields/select_field_with_detail_answer.py

```diff
@@ -28,8 +28,9 @@ def iter_choices(self):
             yield value, label, self.coerce(value) == self.data, detail_answer_id
 
     def pre_validate(self, form):
-        for value, _, _ in self.choices:
-            if value == self.data:
-                break
-        else:
-            raise ValueError(self.gettext("Not a valid choice"))
+        if self.data is not None:
```

Comment:
This is a nuclear option to get tests working so I can see where changes can be made